docs/container-image-layer-caching.md

```diff
@@ -0,0 +1,214 @@
+# Container Image Layer Caching
+
+This document illustrates how container images are securely cached when pulled from Red Hat's container registry (`registry.access.redhat.com`) through the Squid proxy with a custom store-ID helper, demonstrating the caching optimization for Quay-backed CDN requests.
+
+## The Problem
+Container registries like `registry.access.redhat.com` redirect blob requests to CDN URLs (like `cdn01.quay.io`) with temporary credentials in query parameters. These URLs contain signatures and timestamps that make each request unique, preventing effective caching.
+
+## The Solution
+The custom store-ID helper solves this by:
+
+1. **Authorization Verification**: When Squid receives a CDN request, it asks the store-ID helper to process the URL
+2. **Validation Request**: The helper makes its own GET request to the CDN URL to verify the client is authorized (expects 200 OK)  
+3. **Store-ID Computation**: If authorized, the helper strips query parameters and returns a normalized store-ID for caching
+4. **Cache Key Normalization**: Squid uses this normalized store-ID as the cache key, enabling multiple requests for the same blob (even with different credentials) to hit the same cache entry
+
+## Sequence Diagram
+
+Here's an real example of a client pulling the `registry.access.redhat.com/ubi8/ubi8-minimal:latest`
```

Comment:
```suggestion
Here's a real example of a client pulling the `registry.access.redhat.com/ubi8/ubi8-minimal:latest`
```

Reply:
Here we broke the line short (which I prefer), but above we did not. We should make it consistent.